src/searchModal.js

```diff
@@ -419,13 +441,51 @@ export default function searchModalFactory(config) {
      */
     function buildDataModel(data) {
         if (data.settings) {
+            //save availableColumns to memory
+            availableColumns = data.settings.availableColumns;
             // @todo: use the selected columns instead. It can use a promise as it takes place insise a promise chain
-            data.model = columnsToModel(data.settings.availableColumns);
+            data.model = columnsToModel(availableColumns);
```

Comment:
**chore:** As it seems the columns are refined after, and not before as it was foreseen, you could remove the coment above.

**thought:** Looking at the code, I think the function `buildDataModel` should be removed for the benefit of `filterSelectedColumns` which should do the job. Having the model refined, then filtered looks a waste of resources. I'd rather do the filtering then the refining, or even both at once.

Reply:
Yes, right, removed